osdep/threads-posix.h

```diff
@@ -19,6 +19,9 @@
 
 #include <errno.h>
 #include <pthread.h>
+#if HAVE_BSD_THREAD_NAME
+#include <pthread_np.h>
+#endif
```

Comment:
You should probably move `config.h` to the top.

Reply:
Hi! 

Reason? By convention the system #includes come first, moving "config.h" up would break that, and the alphabetical arrangement.

Additionally, this header (pthread_np.h) is only needed to enable the use of NON-POSIX pthread functions within OpenBSD.

This fix allows me to compile the new version of mpv(0.37.0) on the OpenBSD ports without any problems.

PS: I take this opportunity to change the PR and follow the project guidelines regarding them.

Reply:
Alternatively, you could move this check below `config.h`. We seem to have both variations in the code base when looking around. I'm fine with either.